src/main/cpp/src/api/annotation_service.cc

```diff
@@ -0,0 +1,263 @@
+  genomicsdb_pb::ExportConfiguration export_config;
+  export_config.ParseFromString(export_configuration);
+  for(auto i=0; i<export_config.annotation_source_size(); ++i) {
+    auto filename  = export_config.annotation_source(i).filename();
+    if (!TileDBUtils::is_file(filename)) {
+      throw GenomicsDBException(logger.format("Annotation data source {} not found", filename));
+    }
+
+    // If the vcf file is chromosome-specific and the query doesn't cover those chromosomes
+    // then don't bother loading the file.
+    if(!export_config.annotation_source(i).file_chromosomes().empty()) {
+      bool isVcfChromosomeInQuery = false;
+      for(auto j=0; j<export_config.annotation_source(i).file_chromosomes_size(); ++j) {
+        std::string vcf_contig = export_config.annotation_source(i).file_chromosomes(j);
+        if(contigs.find(vcf_contig) != contigs.end()) {
+          // The chromosome-specifc vcf and the query have a chromosome in common
+          isVcfChromosomeInQuery = true;
+          break;
+        }
+      }
+
+      if(!isVcfChromosomeInQuery) {
+      	// Don't load the chromosome-specific VCF 
+        continue;
+      }
+    }
+
+    genomicsdb_htslib_plugin_initialize();
+
+    std::set<std::string> fields;
+    for(auto field : export_config.annotation_source(i).attributes()) {
+      fields.insert(field);
+    }
+
+    std::set<std::string> file_chromosomes;
+    for(auto chromosome: export_config.annotation_source(i).file_chromosomes()) {
+      file_chromosomes.insert(chromosome);
+    }
+
+    m_annotation_sources.emplace_back(filename,
+                                      export_config.annotation_source(i).data_source(),
+                                      fields,
+                                      file_chromosomes);
+  }
+  m_annotation_buffer.reserve(64);
+}
+
+std::vector<annotation_source_t>& AnnotationService::get_annotation_sources() {
+  return m_annotation_sources;
+}
+
+std::unordered_map<int, int> bcf_ht_type_to_nbytes = {
+  {BCF_HT_INT, sizeof(int)},
+  {BCF_HT_INT64, sizeof(long)},
+  {BCF_HT_REAL, sizeof(float)},
+  {BCF_HT_FLAG, 0},
+  {BCF_HT_STR, 1}
+};
+
+/**
+  Create a new genomic_field_t
+ */
+genomic_field_t AnnotationService::get_genomic_field(const std::string &data_source,
+                                                     const std::string &info_attribute,
+                                                     const char *value,
+                                                     const int32_t value_length,
+                                                     const int bcf_ht_type) {
+  std::string genomic_field_name = data_source + "_" + info_attribute;
+  m_annotation_buffer.push_back(std::move(std::string(value, bcf_ht_type_to_nbytes.at(bcf_ht_type)*value_length)));
+  return genomic_field_t(genomic_field_name, m_annotation_buffer.back().data(), value_length);
+}
```

Comment:
Could we directly the string along to genomic_field_t instead of referencing `.back().data()`. It seems like we could get into potential trouble if we require more than 64 annotations here.